009-OFEP-add-dispose.md

```diff
@@ -0,0 +1,45 @@
+## Add dispose functionality to API
+
+## State: DRAFTING
+
+The goal of this OFEP is to enhance the OpenFeature API to allow authors to cleanly dispose of any resources by consumed by OpenFeature providers.
+
+## Background
+
+When implementing OpenFeature providers, you sometimes need to register event handlers, or timers or similar functionality to
+correctly handle functionality of the vendor library. Some vendor libraries might also have analytics, or debugging functionality
+that can be enabled and might get flushed to the vendor platform when the library gets stopped or disposed off.
+
+Currently, you aren't able to cleanly handle these use cases in OpenFeature providers.
+
+## Proposal
+
+To allow to cleanly dispose of resources consumed by OpenFeature providers, I would like to purpose a `dispose`-function that
+can be called by OpenFeature SDK to allow to cleanup these resources. The OpenFeature provider will implement an async function
+named `dispose` that can be called by the internal OpenFeature SDK
+ 
+### [Example] Implementation
+
+The `Provider` type gets extended with the following function:
```

Comment:
I think this should be optional on the provider interface to avoid introducing a breaking change.

Reply:
Yep, agreed. That's why I used a new interface in my example. I will make it clearer in the OFEP though.

Reply:
Done. I was intentionally vague here since there's a number of ways to do this sort of thing and I think that should be left up to the SDK authors.